launchmon/src/linux/sdbg_linux_launchmon.cxx

```diff
@@ -809,6 +811,106 @@ linux_launchmon_t::is_bp_prologue_done (
     }    
 }
 
+//! PRIVATE: linux_launchmon_t::continue_on_attach
+/*!
+    Builds the on-demand proctable if it hasn't been built already.
+
+*/
+bool
+linux_launchmon_t::continue_on_attach (
+                 process_base_t<SDBG_LINUX_DFLT_INSTANTIATION>& p,
+         bool use_cxt )
+{
+  try
+    {
+      using namespace std;
+
+      image_base_t<T_VA,elf_wrapper>* main_im;
+
+      int local_pcount = 0;
+#if MEASURE_TRACING_COST
+      double c_start_ts;
+      double c_end_ts;
+      c_start_ts = gettimeofdayD();
+#endif
+      main_im  = p.get_myimage();
+      if (!main_im)
+        {
+          self_trace_t::trace (
+             true,
+             MODULENAME, 1,
+             "main image not processed");
+          return LAUNCHMON_FAILED;
+        }
+
+      //
+      // fetching the RPDTAB size
+      //
+      symbol_base_t<T_VA> debug_ps
+        = main_im->get_a_symbol ( p.get_launch_proctable_size() );
+
+      if (!debug_ps && p.get_myrmso_image())
+        {
+          debug_ps
+            = p.get_myrmso_image()->get_a_symbol(p.get_launch_proctable_size());
+        }
+
+      T_VA procsize_addr = debug_ps.get_relocated_address();
+
+      get_tracer()->tracer_read( p,
+                 procsize_addr,
+                 &(local_pcount),
+                 sizeof(local_pcount),
+                 use_cxt );
+
+      if (local_pcount < 0 )
+        {
+          self_trace_t::trace (
+             true,
+             MODULENAME, 1,
+             "MPIR_proctable_size is negative");
+          return LAUNCHMON_FAILED;
+        }
+
+      // If the local_pcount is non-zero, the proctable has already been built
+      // and we should not issue the continue command.
+      if (local_pcount > 0)
+        {
+          return LAUNCHMON_FAILED;
+        }
+
+      // Need to continue since the proctable hasn't been build yet.
+      get_tracer()->tracer_continue (p, use_cxt);
+
+#if MEASURE_TRACING_COST
+      c_end_ts = gettimeofdayD();
+      {
+        self_trace_t::trace ( true,
+           MODULENAME,0,
+           "PROCTAB(%d) Continue_on_attach: %f ",
+           local_pcount, (c_end_ts - c_start_ts));
+      }
+#endif
+
+      return LAUNCHMON_OK;
```

Comment:
`true`.... Or maybe it makes more sense to change the function return type to `int`...